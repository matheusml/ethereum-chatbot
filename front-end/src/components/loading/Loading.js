import React from "react";
import { View, ActivityIndicator, StyleSheet } from "react-native";
import PropTypes from "prop-types";

export default class Loading extends React.Component {
  render() {
    const { loading } = this.props;
    if (loading) {
      return (
        <View style={styles.loading}>
          <ActivityIndicator size="large" color="white" />
        </View>
      );
    }
    return null;
  }
}

Loading.propTypes = {
  loading: PropTypes.bool
};

const styles = StyleSheet.create({
  loading: {
    zIndex: 1,
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "black",
    opacity: 0.6
  }
});
