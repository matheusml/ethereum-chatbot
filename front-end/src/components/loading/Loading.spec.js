import React from "react";

import Loading from "./Loading";

import renderer from "react-test-renderer";

describe("Loading", () => {
  it("should match snapshot with loading as true", () => {
    const tree = renderer.create(<Loading loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with loading as false", () => {
    const tree = renderer.create(<Loading loading={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
