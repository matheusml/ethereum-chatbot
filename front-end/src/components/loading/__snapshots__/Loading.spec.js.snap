// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match snapshot with loading as false 1`] = `null`;

exports[`Loading should match snapshot with loading as true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "opacity": 0.6,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
